Added YAML support

nijel opened the pull request:
There are two classes to support two different YAML formats:

YAMLFile : generic key/value format
RubyYAMLFile : Ruby format with target language as root node

Fixes #3248

unho commented on translate/storage/test_yaml.py:
```diff
@@ -0,0 +1,120 @@
+    UnitClass = yaml.YAMLUnit
+
+
+class TestYAMLResourceStore(test_monolingual.TestMonolingualUnit):
```
I guess this should be `test_monolingual.TestMonolingualStore`.

unho replied:
If you change this you will see that several tests fail. Please fix them.

nijel replied:
Oops, I've based the tests on resx and the same issue is there (translate/storage/test_resx.py).

nijel replied:
See #3504 and #3505.

unho replied:
Yes, thanks a lot for those PRs.

unho commented on translate/storage/yaml.py:
```diff
@@ -0,0 +1,230 @@
+
+from translate.storage import base
+
+import yaml.constructor
```
This import should be before translate toolkit one.

unho commented on translate/storage/test_yaml.py:
```diff
@@ -0,0 +1,120 @@
+'''
+
+
+class TestRubyYAMLResourceStore(test_monolingual.TestMonolingualUnit):
```
This one should be `test_monolingual.TestMonolingualStore` too.

unho commented:
Very nice work, but if you correct the two misspelled classes in tests it unveils several failing tests. Please fix that.

unho commented:
On a related note I was able to very quickly code some converters on top of this.

unho commented:
It would be great too if you can further adjust this so comments are extracted. Here you have an example file with comments https://github.com/openstreetmap/openstreetmap-website/blob/master/config/locales/en.yml#L44-L45

nijel commented:
Indeed supporting comments would be great, but I've decided to ignore this part for initial implementation as PyYAML support for this is poor. There is PyYAML fork which should be better (https://pypi.python.org/pypi/ruamel.yaml), but I have no experience with it.

Anyway I've addressed all issues you've pointed out (+ rebased changes on current master).

unho commented:
We can live without comments for now. Landed in master.

rvanlaak commented:
@unho awesome! Any news about when the next beta will be released? And @nijel how can I let Weblate install that specific beta version of `translation-toolkit`? 👍

nijel commented:
You can use ``pip install 'translate-toolkit>=2.0.0b4'`` (or something similar).
